lib/config/ChainConfigManager.ts

```diff
@@ -0,0 +1,296 @@
+import { ChainId } from '@uniswap/sdk-core';
+import { RoutingType } from '../constants';
+
+type ChainConfig = {
+  routingTypes: (
+    |
+    {
+      routingType: RoutingType.CLASSIC | RoutingType.DUTCH_V2 | RoutingType.RELAY,
+      skipRFQ?: boolean,
+      priceImprovementBps?: number,
+      stdAuctionPeriodSecs?: number,
+      deadlineBufferSecs?: number
+    }
+    | 
+    {
+      routingType: RoutingType.DUTCH_LIMIT,
+      skipRFQ?: boolean,
+      priceImprovementBps?: number,
+      stdAuctionPeriodSecs?: number,
+      lrgAuctionPeriodSecs?: number,
+      deadlineBufferSecs?: number
+    }
+  )[],
+  alarmEnabled: boolean
+}
+
+export abstract class ChainConfigManager {
+
+  // Represents the other route dependencies for each route type
+  // If a route is added in to the supported routingTypes for a chain,
+  // all dependencies will also be supported
+  private static readonly _routeDependencies: { [routingType: string] : RoutingType[] } = {
+    [RoutingType.DUTCH_LIMIT]: [
+      RoutingType.CLASSIC
+    ],
+    [RoutingType.DUTCH_V2]: [
+      RoutingType.CLASSIC
+    ]
+  }
+  private static readonly _chainConfigs: { [chainId:number] : ChainConfig } = {
+    [ChainId.MAINNET]: {
+      routingTypes: [
+        {
+          routingType: RoutingType.CLASSIC 
+        },
+        {
+          routingType: RoutingType.DUTCH_LIMIT,
+          lrgAuctionPeriodSecs: 120
+        },
+        {
+          routingType: RoutingType.RELAY 
+        },
+        {
+          routingType: RoutingType.DUTCH_V2,
+          // 10 blocks from now
+          // to cover time to sign, run secondary auction, and some blocks for decay
+          deadlineBufferSecs: 120
+        }
+      ],
+      alarmEnabled: true
+    },
+    [ChainId.OPTIMISM]: {
+      routingTypes: [
+          {
+            routingType: RoutingType.CLASSIC 
+          },
+      ],
+      alarmEnabled: true
+    },
+    [ChainId.OPTIMISM_GOERLI]: {
+      // TODO: add back optimism GOERLI once we are sure routing api supports it
+      routingTypes: [],
+      alarmEnabled: false
+    },
+    [ChainId.ARBITRUM_ONE]: {
+      routingTypes: [
+        {
+          routingType: RoutingType.CLASSIC 
+        },
+        {
```

Comment:
This is new (DutchV2 on Arbitrum) otherwise everything else replicates existing behavior.